jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/DockerContextTask.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle;
+
+import com.google.cloud.tools.jib.builder.BuildConfiguration;
+import com.google.cloud.tools.jib.docker.DockerContextGenerator;
+import com.google.common.base.Preconditions;
+import com.google.common.io.InsecureRecursiveDeleteException;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+
+public class DockerContextTask extends DefaultTask {
+
+  @Nullable private String targetDir;
+  @Nullable private String baseImage;
+  @Nullable private List<String> jvmFlags;
+  @Nullable private String mainClass;
+
+  @Input
+  @Nullable
+  public String getBaseImage() {
+    return baseImage;
+  }
+
+  @Input
+  @Nullable
+  public List<String> getJvmFlags() {
+    return jvmFlags;
+  }
+
+  @Input
+  @Nullable
+  public String getMainClass() {
+    return mainClass;
+  }
+
+  /** The output directory for the Docker context is by default {@code build/jib-dockercontext}. */
+  @OutputDirectory
+  public String getTargetDir() {
+    if (targetDir == null) {
+      return getProject().getBuildDir().toPath().resolve("jib-dockercontext").toString();
+    }
+    return targetDir;
+  }
+
+  /** The output directory can be overriden with the {@code --jib.dockerDir} command line option. */
+  @Option(option = "jib.dockerDir", description = "Directory to output the Docker context to")
+  public void setTargetDir(String targetDir) {
+    this.targetDir = targetDir;
+  }
+
+  @TaskAction
+  public void generateDockerContext() {
+    // Asserts required parameters are not null.
+    Preconditions.checkNotNull(baseImage);
+    Preconditions.checkNotNull(jvmFlags);
+    Preconditions.checkNotNull(mainClass);
+
+    // TODO: Refactor with BuildImageTask.
+    ProjectProperties projectProperties = new ProjectProperties(getProject(), getLogger());
+
+    String mainClass = this.mainClass;
```

Comment:
Seems like we might want to eventually have a mainClassProvider to share this across tasks?